server/server.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	pb "github.com/romanlevin/felek/jobs"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+)
+
+type server struct {
+	pb.UnimplementedJobsServer
+	jobs     map[string]*job
+	logDir   string
+}
+
+func newServer() (*server, error) {
+	logDir, err := ioutil.TempDir("", "felek-logs-*")
+	if err != nil {
+		return nil, err
+	}
+
+	s := &server{jobs: make(map[string]*job), logDir: logDir}
+	return s, nil
+}
+
+func (s *server) storeJob(j *job) {
+	s.jobs[j.id] = j
+}
+
+func (s *server) getJob(id string) (*job, bool) {
+	j, ok := s.jobs[id]
+	return j, ok
+
+}
```

Comment:
Can't these be called concurrently by multiple clients?

Reply:
Yes. I'm protecting access to attributes, not to the job struct itself.

I guess an alternative would be to store job by value, and requiring a lock to put it back, but Cmd has to be stored by reference anyway.

I could also put the job struct in its own package and force the server to access it through methods only.

Reply:
I don't think it's a question about reference vs value, but rather whether the map access is concurrency-safe?

Reply:
Oh I misunderstood the question earlier. 

Turns out the answer is "no, [it is not](https://golang.org/doc/faq#atomic_maps)". For a language so focused on concurrency I'm constantly surprised by decisions around it.

Guess I'm bringing back that server-wide lock to protect map access.

Reply:
Protected access to `jobs` with a `sync.RWMutex`.